### ISSUE TITLE
react-reconciler.development.js:130 Uncaught TypeError: Cannot read property 'alternate' of undefined

### ISSUE THREAD

**Issue by PawelGIX** [lavrton/react-konva#171]
Any ideas what could be wrong?

![image](https://user-images.githubusercontent.com/391153/36078781-5e06e172-0f7b-11e8-9a62-09bbe89d74cf.png)

**Comment by lavrton** [comment]
@PawelGIX how can I reproduce?

**Comment by Domino987** [comment]
@PawelGIX Did you find a fix?

**Comment by PawelGIX** [comment]
I think there was something wrong with my build process. The problem no longer occurs.

**Comment by damiangreen** [comment]
I'm now seeing this after  attempting to upgrade to React 17. Not sure if there is a react-konva fiddle i can try with to repro

**Comment by lavrton** [comment]
I want to wait for official `v17` release to make an update for `react-konva`.

**Comment by AonanLi** [comment]
having this issue with "react": "16.13.1"

**Comment by lavrton** [comment]
@AonanLi what `react-konva` version do you have? They should much and you should not have peer deps warnings.

**Comment by AonanLi** [comment]
> @AonanLi what `react-konva` version do you have? They should much and you should not have peer deps warnings.

I was using 17.0.0. I switched to 16.13.0-6 and it works now. Thanks for pointing that out!

**Comment by jakeei212** [comment]
same issue on "react": "^16.13.1"

**Comment by damiangreen** [comment]
React 17 is now officially released, will this lib be updated?

**Comment by lavrton** [comment]
Already. `npm install react-konva@17.0.0-0`

**Comment by damiangreen** [comment]
jsut to confirm react-konva@17.0.0-0 will only work with react 17 right? I jsut tried to install it into a react 16.x app with the erro above...

Great work on the library

**Comment by lavrton** [comment]
You don't need to install the last `react-konva` it with `react 16.x`. Last `react-konva` update is just internal dependencies updates for compatibility with the last react.

**Comment by derit** [comment]
i'm using react 16.13.1 
to solved this issue 
`yarn add react-conva@^16.13.0-6`

**Comment by casamia918** [comment]
Upgrade react and react-dom package to 17.0.1

```json
{

    "@types/react": "^17.0.0", // for typescript
    "@types/react-dom": "^17.0.0", // for typescript
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-konva": "^17.0.1-3",

}

```

**Comment by flyon** [comment]
@derit that did the trick for me too. Anyone still using react 16.x and adding react-konva with `yarn add react-konva` or `npm install react-konva --save` will currently run into this issue so @lavrton I think a change in package.json to state that the latest version _requires_ react 17, and not anything below it will help with this. 

I think you meant **k**onva not _c_onva.

**Comment by lavrton** [comment]
The `package.json` has: `"react": ">=17.0.1",` in `peerDependencies`. So I guess you had a peer dependencies warning and you just ignore it (as everybody else myself included). Probably I will add runtime warning back.

**Comment by sylvestrevgen** [comment]
@lavrton I have the same issue on React 16 - TypeError: Cannot read property 'alternate' of undefined. I installed react-konva and konva with just yarn add react-konva konva. What can I do to fix this issue?

**Comment by lavrton** [comment]
Upgrade `react` and `react-dom` to the last version (recommended) or downgrade `react-konva` with `npm install react-konva@^16` (not recommended).

**Comment by sylvestrevgen** [comment]
I'm sorry. My problem is that I want to render canvas in
isomorphic application with one codebase for client and server. Can you
advice, please, how can I use canvas in isomorphic app? Or is it
impossible cause server side?

**Comment by sylvestrevgen** [comment]
@lavrton I use React Starter Kit https://github.com/kriasoft/react-starter-kit

**Comment by lavrton** [comment]
> Or is it impossible cause server side?

It is impossible right now. `react-konva` will just render empty div without errors.

**Comment by sylvestrevgen** [comment]
@lavrton  I understood. Thank you. I will not use canvas and will find another solution.

**Comment by NikulGoyani369** [comment]
it's working after updating react-konva npm packages.

**Comment by Siva-at-OFS** [comment]
I got exact problem when I upgraded the react version from 16.3.0 to 17.0.0 and here's how I solved it.

Spoiler alert: Its a combination of version upgrade and minor code change.

Step 1: Upgraded both konva and react-konva to their latest packages (~8.3.0 and ~17.0.2-5 respectively).

Step 2: Tested the application to find a different error but more controllable error (You situation may differ depending on code)
![image](https://user-images.githubusercontent.com/84328892/143984913-9c889172-3fb7-46d2-89be-908c3347fc2c.png)
  
Step 3: Since its not excepting the height and width of canvas element to be set to zero, I changed them to 1 (No significant impact on UI)

Feature is working as expected.
